linkcheck/checker/urlbase.py

```diff
@@ -316,6 +316,10 @@ def set_cache_url (self):
         # remove anchor from cached target url since we assume
         # URLs with different anchors to have the same content
         self.cache_url = urlutil.urlunsplit(self.urlparts[:4]+[u''])
+        if self.anchor:
+            # and bring anchor (now to full url) back since otherwise those
+            # urls would not be considered e.g. by AnchorCheck plugins
+            self.cache_url += '#%s' % self.anchor
```

Comment:
hmm... i guess i didn't realize this would invalidate the cache... we should definitely cache the page only once per anchor.

maybe the problem is in the AnchorCheck plugin?

Reply:
I believe it doesn't even see some of those as an analysis in the #179 by others showed

Reply:
I think this is close, and only needed if using AnchorCheck.